lib/chef/client.rb

```diff
@@ -715,11 +718,57 @@ def self.generate_pfx_package(cert_name, date)
       pfx
     end
 
-    def create_new_key_and_register(cert_name)
+    def update_key_and_register(cert_name)
+      # Chef client and node objects exist on Chef Server already
+      # Create a new public/private keypair in secure storage
+      # and register the new public cert with Chef Server
       require "time" unless defined?(Time)
       autoload :URI, "uri"
 
-      # KeyMigration.instance.key_migrated = true
+      node = Chef::Config[:node_name]
+      d = Time.now
+      if d.month == 10 || d.month == 11 || d.month == 12
+        end_date = Time.new(d.year + 1, d.month - 9, d.day, d.hour, d.min, d.sec).utc.iso8601
```

Comment:
I think we can also simplify with something like `Time.new(...) + (3600  * 24 * 90)` for 90 days.

Reply:
corrected.